search/includes/classes/class-health.php

```diff
@@ -261,6 +262,10 @@ public function validate_index_posts_content( $start_post_id, $last_post_id, $ba
 		if ( $process_parallel_execution_lock && $this->is_validate_content_ongoing() ) {
 			return new WP_Error( 'content_validation_already_ongoing', 'Content validation is already ongoing' );
 		}
+		$interrupted_start_post_id = $this->get_validate_content_abandoned_process();
+		if ( $process_parallel_execution_lock && $interrupted_start_post_id ) {
```

Comment:
Should we be checking if a `$start_post_id` was actually passed into `validate_index_posts_content()`? Right now you could call the command with this both `--start_post_id` and `--force-parallel-execution` and it would ignore your `--start_post_id` if processing had been interrupted. Is that the desired outcome, or do we want to allow it to be overridden from the command?

Reply:
Hmm the naming isn't great here really... `$process_parallel_execution_lock = ! $force_parallel_execution`. I didn't want to have the negation everywhere to not be confusing so I tried to introduce this, I thought descriptive, variable.

So actually when you send in `--force-parallel-execution` the `$process_parallel_execution_lock === false` and then this `if` will be false and the `$interrupted_start_post_id` will be ignored.

However when you **don't** send in `--force-parallel-execution`, the  `$interrupted_start_post_id` will take precedence (and `--start_post_id` would be ignored).

I did think about skipping process tracking if `$start_post_id > 1` or `$last_post_id` is send in. But then I thought it would be a valid usecase... say it always dies on post id 350001, so you would say start from 340000 and last_id 350000 (but track and clear process so that we could do more work later). And in general it felt like there might be more complicated combinations of parameters when to track and when not.

So I thought, let's keep it simple. Sending `--force_parallel_execution` ignore the lock and ignores the tracking. Hmm maybe the parameter should be just `--force` then? What do you think?

Reply:
Hmm interesting - IMO the `--start_post_id` flag should take precedence - if it's passed, respect the user's wishes and start there. It would be pretty confusing to pass in the post id you want to start at and have it actually start at another seemingly random location. At least if we're overriding the user's choice, we need to tell them, and tell them how to stop that behavior (by using `--force_parallel_execution`.

Reply:
I added the change discussed in slack.